src/scalar.rs

```diff
@@ -540,6 +543,80 @@ impl Zeroize for Scalar {
     }
 }
 
+#[cfg(feature = "elliptic-curve")]
+impl group::ff::Field for Scalar {
+    fn random(mut rng: impl RngCore) -> Self {
+        let mut scalar_bytes = [0u8; 64];
+        rng.fill_bytes(&mut scalar_bytes);
+        Self::from_bytes_mod_order_wide(&scalar_bytes)
+    }
+
+    fn zero() -> Self {
+        Self::zero()
+    }
+
+    fn one() -> Self {
+        Self::one()
+    }
+
+    fn square(&self) -> Self {
+        self * self
+    }
+
+    fn double(&self) -> Self {
+        self + self
+    }
+
+    fn invert(&self) -> subtle::CtOption<Self> {
+        let result = self.invert();
+        subtle::CtOption::new(result, !self.is_zero())
+    }
+
+    /// Unimplemented!
+    fn sqrt(&self) -> subtle::CtOption<Self> {
+        unimplemented!()
+    }
+}
+
+#[cfg(feature = "elliptic-curve")]
+impl group::ff::PrimeField for Scalar {
+    type Repr = GenericArray<u8, U32>;
+
+    fn from_repr(repr: Self::Repr) -> subtle::CtOption<Self> {
+        let s = Scalar { bytes: repr.into() };
+        // Ensure that s < 2^255 by masking the high bit
+        let valid = subtle::ConstantTimeLess::ct_lt(&s.bytes[31], &0b0111_1111);
+        subtle::CtOption::new(s, valid)
+    }
+
+    fn to_repr(&self) -> Self::Repr {
+        self.to_bytes().into()
+    }
+
+    fn is_odd(&self) -> Choice {
+        self.bytes[0].ct_eq(&1)
+    }
+
+    const NUM_BITS: u32 = 253;
+
+    const CAPACITY: u32 = Self::NUM_BITS - 1;
+
+    fn multiplicative_generator() -> Self {
+        7_u8.into()
+    }
```

Comment:
I think this might be wrong? In Sage I get two:

```sage
sage: n = 2**252 + 27742317777372353535851937790883648493
sage: GF(n).primitive_element()
2
```